### PR TITLE
create entity codes table

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/models/entitycode.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"github.com/gofrs/uuid"
+
+	"github.com/silinternational/cover-api/api"
+)
+
+type EntityCodes []EntityCode
+
+type EntityCode struct {
+	ID   uuid.UUID `db:"id"`
+	Code string    `db:"code"`
```
Comment (Baggerone) [inline application/models/entitycode.go:17]: Do you want to include any validation?

File: application/models/entitycode.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	}
+
+	if perm == PermissionList {
+		return true
+	}
+
+	return false
+}
+
+func (ec *EntityCodes) ConvertToAPI(tx *pop.Connection) interface{} {
```
Comment (Baggerone) [inline application/models/entitycode.go:53]: Does this need to be an interface return value?

**Review by Baggerone — APPROVED** [review]

**Review by fillup — APPROVED** [review]

File: application/models/entitycode.go (hunk 2 above)
Comment (Schparky) [reply application/models/entitycode.go:53]: Oops, I'll bet Goland did that when I used the "create method" helper.

File: application/models/entitycode.go (hunk 1 above)
Comment (Schparky) [reply application/models/entitycode.go:17]: Hmm, maybe not since it may never be a user-created data type.

**Pull request merged**